[jonyeokj] iP

jonyeokj opened the pull request:
# Martin Bot

> Martin learned chess so that he could play with his young kids. He still beats them most of the time. ([source](https://www.chess.com/member/martin-bot))

Once a chess bot, now a chatbot!

Martin is:

- 🖋 Text-based
- 🎓 Easy to pick up
- 🚄 ~~FAST~~ Super FAST to use

All you need to do is,

1. download martin [here](https://github.com/jonyeokj/ip/releases/download/A-Jar/martin.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you!

## 🎉 It's Absolutely **Free**! 🎉

**Features:**
- [x]  Managing tasks
- [ ] Managing deadlines (_coming soon_)
- [ ] Reminders (_coming soon_)

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

jonyeokj committed:
Changed Duke to Martin

jonyeokj committed:
Added isDone Functionality and Task.java

jonyeokj committed:
Added Tracking for Tasks: Todos, Events, Deadlines

jonyeokj committed:
Abstraction of adding Task functions

jonyeokj committed:
Edited Duke to Martin in runtest files

jonyeokj committed:
Added Exceptions for empty tasks descriptions, invalid commands, invalid task numbers and already marked/unmarked tasks

jonyeokj committed:
Added Delete Function for tasks

jonyeokj committed:
Added saving functionality, tasks are saved and remembered

jonyeokj committed:
Removed wildcard import and added header messages to functions

jonyeokj committed:
Added Date and Time formatting to Deadline class, also Added date function in main

jonyeokj committed:
Created Parser, Storage, UI and TaskList classes for better OOP

jonyeokj committed:
Added Command interface and classes and organized files into packages

jonyeokj committed:
Added Gradle Support

jonyeokj committed:
Added DeleteCommandTest and TodoCommandTest files

jonyeokj committed:
Implemented the Fat Jar

jonyeokj committed:
Add JavaDoc comments to the code

jonyeokj committed:
Added Find Command

jonyeokj committed:
Added Header comments for Exception classes

Gavino3o commented on src/main/java/martin/parser/Parser.java:
```diff
@@ -0,0 +1,72 @@
+            throw new InvalidCommandException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+
+        String[] parts = input.split(" ", 2);
```
Good use of nouns and paid special attention to using plural forms when needed. Good job!

Gavino3o commented on src/main/java/martin/parser/Parser.java:
```diff
@@ -0,0 +1,72 @@
+                return new DateCommand(input, tasks);
+            case "find":
+                return new FindCommand(input, tasks);
+            default:
```
Default case included as per coding standard. 👍

Gavino3o commented on src/main/java/martin/task/Event.java:
```diff
@@ -0,0 +1,17 @@
+
+    @Override
+    public String toString() {
+        return "[E][" + (isDone ? "X" : " ") + "] " + description + " (from: " + from + " to: " + to + ")";
```
Consider using String.format here, it makes the string more readable and easier to edit if required in the future.

Gavino3o commented on src/main/java/martin/task/Task.java:
```diff
@@ -0,0 +1,94 @@
+    
+        String type = parts[0];
+    
+        switch (type) {
```
If it's only two conditions, consider using if else instead. But I can see the potential of including todos, events and more subtypes of Tasks in the future.

Gavino3o commented on src/main/java/martin/Ui.java:
```diff
@@ -0,0 +1,37 @@
+package martin;
+import java.util.Scanner;
+
```
Consider adding JavaDocs for classes for a short description of what the class encapsulates or can do.

Gavino3o reviewed (commented):
Overall, the PR LGTM 👍. Some small changes can be considered but good progress has been made! Cheers and have a good day.

wui-hong commented on src/main/java/martin/task/Task.java:
```diff
@@ -0,0 +1,94 @@
+        isDone = true;
+    }
+
+    public void unmarkAsDone() {
```
Would it be more appropriate to rename the function to `markAsUndone`?

wui-hong commented on src/main/java/martin/Ui.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    public void showLine() {
+        System.out.println("    ____________________________________________________________");
```
Perhaps adding the Line string as a constant would make your code more readable and reusable

wui-hong commented on src/main/java/martin/Storage.java:
```diff
@@ -0,0 +1,74 @@
+                Files.createFile(path);
+            }
+            Files.write(path, lines, StandardOpenOption.TRUNCATE_EXISTING);
+        } catch (IOException e) {
```
Good use of error handling throughout the code

wui-hong commented on src/main/java/martin/commands/DateCommand.java:
```diff
@@ -0,0 +1,61 @@
+        }
+
+        ui.showLine();
+        System.out.println("     Tasks on " + date.format(DateTimeFormatter.ofPattern("M d yyyy")) + ":");
```
Might be neater to split out the date formatting section onto a new line

wui-hong reviewed (commented):
Overall, LGTM. Only minor suggestions but they are nitpicks!

jonyeokj committed:
Changed Martin Bot from CLI to a GUI interface

erohsikivar commented on src/main/java/martin/exceptions/InvalidCommandException.java:
```diff
@@ -0,0 +1,12 @@
+ *
+ * @param message Error message to be printed out.
+ */
+public class InvalidCommandException extends MartinException {
```
I feel maybe don't so many classes for each exceptions since all your class methods are implementing same line of code super(message).

erohsikivar commented on src/main/java/martin/gui/DialogBox.java:
```diff
@@ -0,0 +1,63 @@
+package martin.gui;
```
Good to see that you have separate GUI package.

erohsikivar reviewed (commented):
Overall LGTM.  Just some minor things can be changed.

jonyeokj committed:
Implement assertions for null values

- Added assertions to handle null values for input in the MainWindow class.
- Ensured null responses are appropriately handled in the Martin class.
- Introduced error handling for scenarios where the input is empty.

jonyeokj committed:
Merge pull request #1 from jonyeokj/branch-A-Assertions

Implement assertions for null values

jonyeokj committed:
Implement A-CodeQuality

Refactored code in the DateCommand class to minimize arrowhead style code. Also added javadoc headers for all the command messages briefly explaining what theydo, and cleaned up the headers for their respective execute() commands.

jonyeokj committed:
Merge pull request #2 from jonyeokj/branch-A-CodeQuality

Implement A-CodeQuality

jonyeokj committed:
Add Java CI workflow using GitHub Actions

jonyeokj committed:
Changed runtest files

jonyeokj committed:
Change JDK version

jonyeokj committed:
Add LocalDateTime support to Events

jonyeokj committed:
Implement Snooze and enhance Event format

This commit introduces the snooze command for the task management app.
Users can now snooze their tasks, delaying them by a specified amount
of times in minutes.

Additionally, events now support the toFileFormat method.

Javadoc headers have also been added to all command class constructors.

jonyeokj committed:
Merge pull request #4 from jonyeokj/branch-B-Snooze

Implement Snooze and enhance Event format

jonyeokj committed:
Implement Sort Command and support for parsing Events

Implement SortCommand to enable task sorting functionality.
Extend the fromFileFormat method in Task to handle Event tasks.

jonyeokj committed:
Merge pull request #5 from jonyeokj/branch-C-Sort

Implement Sort Command and support for parsing Events

jonyeokj committed:
Improve GUI and updated README

jonyeokj committed:
Edit README file

jonyeokj committed:
Replace gradle.yml file

jonyeokj committed:
Change PATH in runtest files

jonyeokj committed:
Replace gradle.yml file

jonyeokj committed:
Update FindCommand and Add NoTaskFoundException classes

jonyeokj committed:
Remove text-ui-test file

jonyeokj committed:
Change build.gradle file
